pkgs/by-name/nm/nmgui/package.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  wrapGAppsHook4,
+  gobject-introspection,
+  gtk4,
+  glib,
+  makeDesktopItem,
+  copyDesktopItems,
+  networkmanager,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "nmgui";
+  version = "1.0.0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "s-adi-dev";
+    repo = "nmgui";
+    tag = "v${version}";
+    hash = "sha256-HS/n40Ng8S5N14DtEH/upwlxdzwCoOEJA40EMdCcLw4=io";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    gobject-introspection
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    gtk4
+    glib
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "nmgui";
+      exec = "nmgui";
+      icon = "network-wireless-symbolic";
+      comment = "GTK4-based Network Manager GUI using nmcli";
+      desktopName = "NM GUI";
+      categories = [
+        "Network"
+        "GTK"
+      ];
+      startupNotify = true;
+    })
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/applications,opt/nmgui}
+    # Copy the app files
+    cp -r app $out/opt/nmgui/
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${python3Packages.python}/bin/python3 $out/bin/nmgui \
+      --add-flags "$out/opt/nmgui/app/main.py" \
+      "''${gappsWrapperArgs[@]}"
+  '';
```

you should reference the interpreter through nix, that will be more robust e.g. when building against pypy or against python4 if that ever exists.
You should also add the PYTHONPATH here, or else everything breaks.

```suggestion
  postFixup = ''
    makeWrapper ${python3Packages.python.interpreter} $out/bin/nmgui \
      --prefix PYTHONPATH : "$PYTHONPATH" \
      --add-flags "$out/opt/nmgui/app/main.py" \
      ''${gappsWrapperArgs[@]}
  '';
```